src/rez/utils/pip.py

```diff
@@ -0,0 +1,689 @@
+            else:
+                result_requires.append(rez_req)
+
+    # prefix variant with system requirements
+    sys_variant_requires = []
+
+    if "platform" in sys_requires:
+        sys_variant_requires.append("platform-%s" % _system.platform)
+
+    if "arch" in sys_requires:
+        sys_variant_requires.append("arch-%s" % _system.arch)
+
+    if "os" in sys_requires:
+        sys_variant_requires.append("os-%s" % _system.os)
+
+    if "python" in sys_requires:
+        # Add python variant requirement. Note that this is always MAJOR.MINOR,
+        # because to do otherwise would mean analysing any present env markers.
+        # This could become quite complicated, and could also result in strange
+        # python version ranges in the variants.
+        #
+        sys_variant_requires.append("python-%s" % str(python_version.trim(2)))
+
+    return {
+        "requires": result_requires,
+        "variant_requires": sys_variant_requires + result_variant_requires,
+        "metadata": {
+            "is_pure_python": is_pure_python
+        }
+    }
+
+
+def convert_distlib_to_setuptools(installed_dist):
+    """Get the setuptools equivalent of a distlib installed dist.
+
+    Args:
+        installed_dist (`distlib.database.InstalledDistribution`: Distribution
+            to convert.
+
+    Returns:
+        `pkg_resources.DistInfoDistribution`: Equivalent setuptools dist object.
+    """
+    path = os.path.dirname(installed_dist.path)
+    setuptools_dists = pkg_resources.find_distributions(path)
+
+    for setuptools_dist in setuptools_dists:
+        if setuptools_dist.key == installed_dist.key:
+            return setuptools_dist
+
+    return None
+
+
+def get_marker_sys_requirements(marker):
+    """Get the system requirements that an environment marker introduces.
+
+    Consider:
+
+        'foo (>1.2) ; python_version == "3" and platform_machine == "x86_64"'
+
+    This example would cause a requirement on python, platform, and arch
+    (platform as a consequence of requirement on arch).
+
+    See:
+    * vendor/packaging/markers.py:line=76
+    * https://www.python.org/dev/peps/pep-0508/#id23
+
+    Args:
+        marker (str): Environment marker string, eg 'python_version == "3"'.
+
+    Returns:
+        List of str: System requirements (unversioned).
+    """
+    _py = "python"
+    _plat = "platform"
+    _arch = "arch"
+    _os = "os"
+
+    sys_requires_lookup = {
+        # TODO There is no way to associate a python version with its implementation
+        # currently (ie CPython etc). When we have "package features", we may be
+        # able to manage this; ignore for now
+        "implementation_name": [_py],
+        "implementation_version": [_py],
+        "platform_python_implementation": [_py],
+        "platform.python_implementation": [_py],
+        "python_implementation": [_py],
+
+        "sys.platform": [_plat],
+        "sys_platform": [_plat],
+
+        "os.name": [_os],
```

Comment:
Thinking about it even more, `os.name` can return `posix` (Linux, OSX, etc), `nt` (Windows) or `java` (Jython). We can't do much about Jython, but for the others, it's basically the platform, not the OS (OS would add a dependency on the arch).

Reply:
> Do you know of a reference anywhere that lists valid os.name values? I'll
update this with a comment.

They are, here: https://docs.python.org/3/library/os.html#os.name
https://docs.python.org/3/library/platform.html#platform.system


These are the values I know that are mentioned in the docs.